shared/featureSwitches.ts

```diff
@@ -27,4 +33,27 @@ export const featureSwitches: Record<string, boolean> = {
 	cancellationProductSwitch: false,
 	accountOverviewNewLayout: false,
 	productSwitching: true,
+	savedArticles: false,
+};
+
+// Cypress Testing:
+// We want to be able to write tests independent of switch state that test both states.
+// Use getFeatureSwitches to retrieve switches dependant on clientside environment.
+// For example you can set a switch to false
+// and opt in in Cypress via query search params in cy.visit(url).
+declare global {
+	interface Window {
+		Cypress: unknown;
+	}
+}
+
+const cypressSwitchState: Record<string, boolean> = {
+	...featureSwitches,
+	savedArticles: false,
+};
+export const getFeatureSwitches = (): Record<string, boolean> => {
+	// If running within cypress, return the forced switches to keep tests consistent regardless of production state
```

Comment:
Ah I see, so `window.Cypress` is written to when the tests run. 

Could potentially write the actual switch state on the `window.Cypress` and read this here to return the switches, rather than hardcoding `cypressSwitchState` above. There's an example of that here: https://github.com/guardian/dotcom-rendering/blob/main/dotcom-rendering/cypress/e2e/parallel-1/sign-in-gate.cy.js#L141

Then we can be explicit in the test that `savedArticles` is being set to false.

Reply:
Would be good to validate with someone with a more expert JS knowledge than me whether the use of a global object here is ok

Reply:
For review context, I was basing this solution on the [dotcom solution here](https://github.com/guardian/dotcom-rendering/blob/main/dotcom-rendering/src/web/experiments/cypress-switches.ts)

Reply:
Ah I see, yeah makes sense. I think my point is perhaps a bit of overkill.